modules/stochastic_tools/src/samplers/ParallelSubsetSimulation.C

```diff
@@ -178,13 +182,21 @@ ParallelSubsetSimulation::computeSample(dof_id_type row_index, dof_id_type col_i
   else
   {
     const dof_id_type loc_ind = row_index - getLocalRowBegin();
-    const Real rv = Normal::quantile(getRand(seed_value), _markov_seed[col_index][loc_ind], 1.0);
-    const Real acceptance_ratio = std::log(Normal::pdf(rv, 0, 1)) -
-                                  std::log(Normal::pdf(_markov_seed[col_index][loc_ind], 0, 1));
-    const Real new_sample = acceptance_ratio > std::log(getRand(seed_value + 1))
-                                ? rv
-                                : _markov_seed[col_index][loc_ind];
-    val = Normal::cdf(new_sample, 0, 1);
+    const Real rnd1 = getRand(seed_value);
+    const Real rnd2 = getRand(seed_value + 1);
+
+    if (_sampling_method == "ComponentWiseMH")
```

Comment:
You should keep string comparisons to a minimum and use an enum cakes here like we do everywhere else.